### ISSUE TITLE
Suddenly composer update with package needs --no-dev to skip Dev dependencies installation

### ISSUE THREAD

**Issue by Pherserk** [composer/composer#11705]
My `composer.json`:

E.g. Contains phpunit in require-dev section

When I run this command: 
compose update package/name

Downloads and locks Dev dependencies 

And I expected this to happen:

Dev dependencies to not be installed

It seems --no-dev option became not optional to avoid Dev dependencies installation, sincr today.

Pratical problem:
E.g..
This break any existing deploy pipeline relying on old behaviour.

**Comment by fredden** [comment]
@Pherserk thanks for your report. It sounds like you're saying that there was a change in behaviour between some versions of Composer, but I can't see which versions you're talking about. Please can you share what version you were using that had the old behaviour, and what version you're now using which has the new behaviour?

**Comment by Seldaek** [comment]
Yeah dev has been the default for a long long time https://github.com/composer/composer/commit/caf26ac37c09c6b99ba14db39bb45b7b87cab0e5 - so i am not sure what you are referring to here, unless you've just come out of cryo storage after 10years 😉

**Comment by Pherserk** [comment]
Yeah Jordi in the maximum respect I love composer and thank you any day but this should a Major update.
Legit and needed, web love xi sistency, but a Major.

**Comment by Pherserk** [comment]
Ok not ten years but maybe some months I will better check what my pipeline was downloading.

**Comment by Pherserk** [comment]
It happened but I'm convinced that now Is better better, and easy to avoid,  and that composer version must be fixed in any serious pipeline. 
Lesson learned.

**Comment by Pherserk** [comment]
It happened but I'm convinced that now Is better better, and easy to avoid, and that composer version must be fixed in any serious pipeline.
Lesson learned.
